webpack.config.js

```diff
@@ -24,165 +23,86 @@ const { dependencies } = require( './package' );
 
 const WORDPRESS_NAMESPACE = '@wordpress/';
 
-/**
- * Given a string, returns a new string with dash separators converted to
- * camelCase equivalent. This is not as aggressive as `_.camelCase` in
- * converting to uppercase, where Lodash will also capitalize letters
- * following numbers.
- *
- * @param {string} string Input dash-delimited string.
- *
- * @return {string} Camel-cased string.
- */
-function camelCaseDash( string ) {
-	return string.replace(
-		/-([a-z])/g,
-		( match, letter ) => letter.toUpperCase()
-	);
-}
-
 const gutenbergPackages = Object.keys( dependencies )
 	.filter( ( packageName ) => packageName.startsWith( WORDPRESS_NAMESPACE ) )
 	.map( ( packageName ) => packageName.replace( WORDPRESS_NAMESPACE, '' ) );
 
-const externals = {
-	react: 'React',
-	'react-dom': 'ReactDOM',
-	moment: 'moment',
-	jquery: 'jQuery',
-	lodash: 'lodash',
-	'lodash-es': 'lodash',
+config.entry = gutenbergPackages.reduce( ( memo, packageName ) => {
```

Comment:
I think for the purpose it serves, it shouldn't prove to be an issue, but it's generally poor form to be mutating the exported value of a node dependency, as it will impact any other loaded file which imports the same reference.

Reply:
Agreed, it's a non-issue at the moment but it might become one in the future. Fun fact is that at some point of this PR I added an example of how to use such shared config:

https://github.com/WordPress/gutenberg/blob/cc5cb3613607bbe721533f31de7679b28466a3a3/packages/webpack-config/README.md

I used `Object.assign` to extend the default object :) Somehow I didn't follow my own recommendation. I will open a follow-up PR to improve this code to use good practice 👍

Reply:
PR is ready: #14039.